### PR TITLE
Move to latest code style analyzer package

### PR DESCRIPTION
**Pull request by mavasani**
Also fix the IDExxxx violations from the new package.

Recommended to review commit-by-commit

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/CSharpSourceGeneratorVerifier`1+Test.cs [hunk 1]
```diff
@@ -31,28 +31,17 @@ namespace Microsoft.CodeAnalysis.ResxSourceGenerator.Test
...
 
-            public Test(string identifier, [CallerFilePath] string? testFile = null, [CallerMemberName] string? testMethod = null)
-            {
-                _identifier = identifier;
-                _testFile = testFile;
-                _testMethod = testMethod;
-
-#if WRITE_EXPECTED
-                TestBehaviors |= TestBehaviors.SkipGeneratedSourcesCheck;
-#endif
```
Comment (drewnoakes) [inline src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/CSharpSourceGeneratorVerifier`1+Test.cs]: Looks like the conversion to primary constructors nuked this code.

File: src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/VisualBasicSourceGeneratorVerifier`1+Test.cs [hunk 2]
```diff
@@ -30,28 +30,17 @@ namespace Microsoft.CodeAnalysis.ResxSourceGenerator.Test
...
 
-            public Test(string identifier, [CallerFilePath] string? testFile = null, [CallerMemberName] string? testMethod = null)
-            {
-                _identifier = identifier;
-                _testFile = testFile;
-                _testMethod = testMethod;
-
-#if WRITE_EXPECTED
-                TestBehaviors |= TestBehaviors.SkipGeneratedSourcesCheck;
-#endif
```
Comment (drewnoakes) [inline src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/VisualBasicSourceGeneratorVerifier`1+Test.cs]: Ditto (and possibly elsewhere).

File: src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/CSharpSourceGeneratorVerifier`1+Test.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/CSharpSourceGeneratorVerifier`1+Test.cs]: Oh you address this later in fa45598bd6d27daa3a0bd6e98e021714d3751a2e. Is this a bug for Roslyn?

Comment (mavasani) [reply src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Verifiers/CSharpSourceGeneratorVerifier`1+Test.cs]: Yes, fixed by @CyrusNajmabadi in https://github.com/dotnet/roslyn/pull/68612 yesterday.

**Review by sharwell — CHANGES_REQUESTED** [review]
Please disable IDE0290 and revert the associated changes

**Comment by mavasani** [comment]
> Please disable IDE0290 and revert the associated changes. The rest looks fine.

Tagging @CyrusNajmabadi as well. I thought the idea is to dogfood the new analyzer/fixer for primary constructors. Are you suggesting the revert due to style preferences or the fact that there could still be bugs in the new feature/analyzer/fixer?

**Comment by CyrusNajmabadi** [comment]
I would def appreciate dogfooding.  But if you do run into issues, don't fix them, just revert that case and let me know. :-)

**Comment by mavasani** [comment]
@sharwell Reverted the primary constructor style changes. Please take a look.

**Review by sharwell — APPROVED** [review]

**Pull request merged**